### PR TITLE
Added fixed_string::c_str() and related documentation

### PR DESCRIPTION
**Pull request by DymOK93**
I'm not sure if tests for `data()` and `c_str()` are needed. What do you think about this?

### CODE REVIEW

File: documentation/docs/api/member-functions/data.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+is valid and the values in it correspond to the values stored in the string. 
+
+The returned array is null-terminated, that is, data() and c_str() perform the same function.
+
+## Parameters
+(none)
+
+## Return value
+A pointer to the underlying character storage.
+`c_str() + i == std::addressof(operator[](i))` for every i in [0, size()].
```
Comment (unterumarmung) [inline documentation/docs/api/member-functions/data.mdx:32]:
```suggestion
`c_str() + i == std::addressof(operator[](i))` for every `i` in `[0, size()]`.
```

Comment (unterumarmung) [inline documentation/docs/api/member-functions/data.mdx:32]:
```suggestion
`c_str() + i == std::addressof(operator[](i))` for every i in [0, size()).
```

Maybe?

File: documentation/docs/api/member-functions/c_str.mdx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+</CppOverloadList>
+
+Returns a pointer to a null-terminated character array with data equivalent to those stored in the string. 
+
+## Parameters
+(none)
+
+## Return value
+Pointer to the underlying character storage. 
+`c_str() + i == std::addressof(operator[](i))` for every i in [0, size()].
```
Comment (unterumarmung) [inline documentation/docs/api/member-functions/c_str.mdx:25]:
```suggestion
`c_str() + i == std::addressof(operator[](i))` for every `i` in `[0, size()]`.
```

Comment (unterumarmung) [reply documentation/docs/api/member-functions/c_str.mdx:25]: Ditto: the `)` in `[0, size()]`

File: documentation/docs/api/member-functions/data.mdx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    <CppOverload num={1} code={Overload1} />
+    <CppOverload num={2} code={Overload2} />
+</CppOverloadList>
+
+Returns a pointer to the underlying array serving as character storage. 
+
+The pointer is such that the range `[data(); data() + size()]` 
+is valid and the values in it correspond to the values stored in the string. 
+
+The returned array is null-terminated, that is, data() and c_str() perform the same function.
```
Comment (unterumarmung) [inline documentation/docs/api/member-functions/data.mdx:25]:
```suggestion
The returned array is null-terminated, that is, `data()` and `c_str()` perform the same function.
```

File: documentation/docs/api/member-functions/data.mdx (hunk 1 above)
Comment (DymOK93) [reply documentation/docs/api/member-functions/data.mdx:32]: See http://eel.is/c++draft/strings#basic.string.general-3.sentence-1

Comment (unterumarmung) [reply documentation/docs/api/member-functions/data.mdx]: Ok

**Comment by unterumarmung** [comment]
This PR partially resolves #35 
Thanks!

**Pull request merged**